### ISSUE TITLE
[bug]DocxWriter: Alignment problem with "LineBreak" in any docx produced by pandoc

### ISSUE THREAD

**Issue by ilcpm** [jgm/pandoc#7240]
We are working on a filter about docx, and find some bugs with pandoc due to tags missing in `settings.xml`

## bug reproduce

Operate in any `.docx` file produced by pandoc like this

![image](https://user-images.githubusercontent.com/36830343/115870264-93088180-a471-11eb-862b-9a826fd46482.png)

Type something with **spaces** and use `shift` + `enter` to produce a "**LineBreak**"(choose "show Editing marks" to show the **spaces** in document can be helpful to locate the problem)

Then change the alignment to "Justify" (`Ctrl`+ `J`)

![image](https://user-images.githubusercontent.com/36830343/115870547-fa263600-a471-11eb-8afd-0e27d70f070e.png)

You will find that the words in the first line are so scattered.

![image](https://user-images.githubusercontent.com/36830343/115871164-c39ceb00-a472-11eb-8028-7654459d864e.png)

This problem won't happen in a normal document produced by Word itself.

## how to fix

We unpacked the docx file, compared it with the normal one, and finally found how this problem had happened: there missed a tag `<w:doNotExpandShiftReturn/>` in file `word/settings.xml` tag `<w:compat>`.

![image](https://user-images.githubusercontent.com/36830343/115871810-9bfa5280-a473-11eb-8336-3297cd5cfd3d.png)

This problem was fixed by adding this tag.

![image](https://user-images.githubusercontent.com/36830343/115872020-e4197500-a473-11eb-8d85-99a797048d6d.png)

**Comment by ilcpm** [comment]
This problem will happen in any `.docx` file any case produced by pandoc, unless the missing tag `<w:doNotExpandShiftReturn/>` was added.

If change the alignment of the styles to "Justify" in the reference docx, this problem will directly show up in the output `.docx` file without editing the output file manually.

**Comment by jgm** [comment]
I'm not sure about including this in the default reference.docx.

Note, first, that this doesn't make any difference given pandoc's default settings. It only makes a difference when your document selects justified text alignment (which isn't pandoc's default).

So, do people who enable justification want lines with hard breaks to be justified or not?  It seems to me that this depends on what they use the hard breaks for. One use for a hard break is to fix bad automatic wrapping, and if that's what you use it for, you DO want justification to continue.  Perhaps there are other uses, but I'd suggest that if you want the other behavior in a paragraph, it would make sense to select left-aligned rather than justified.

**Comment by ilcpm** [comment]
Indeed, I don't know in what condition will people use 'LineBreak', but I can't find any GUI command in Word(365 version) to change this setting(add or delete the tag `<w:doNotExpandShiftReturn/>`)

So, if someone use "LineBreak", he will find it's different from the usual document but can't find any settings to change this, that's how we found this problem.

Or, if there really exists this settings in Word can add or delete the tag(`<w:doNotExpandShiftReturn/>`)?

I think, only if there exists this GUI setting can it be kept "not added" in pandoc. If this tag can not be add in Word normally, and can't have any method to add it in pandoc, why it should be missing?

**Comment by ilcpm** [comment]
Sorry, I finally found the compatibility settings in "advanced settings".

![image](https://user-images.githubusercontent.com/36830343/116033114-6df35900-a693-11eb-80ec-d208e397985d.png)

Maybe it should be considered whether to inherit this setting from the "reference-doc". In Chinese, this settings is default.

We found many settings which can be changed in Word GUI which pandoc does not inherit, such as the equation font setting in `settings.xml`.
